Assets/GothicVR/Scripts/Manager/LoadingManager.cs

```diff
@@ -0,0 +1,96 @@
+using UnityEngine.UI;
+using System.Collections.Generic;
+using System.Linq;
+using System;
+
+namespace GVR.Manager
+{
+    public class LoadingManager : SingletonBehaviour<LoadingManager>
+    {
+
+        public enum LoadingProgressType
+        {
+            WorldMesh,
+            VOb,
+            NPC
+        }
+        private GameObject bar;
+
+        private Scene loadingScene;
+
+        private const string loadingSceneName = "Loading";
+
+        private Dictionary<LoadingProgressType, float> progressByType = new Dictionary<LoadingProgressType, float>();
+
+        private void Start()
+        {
+            // Initializing the Dictionary with the default progress (which is 0) for each type
+            foreach (LoadingProgressType progressType in Enum.GetValues(typeof(LoadingProgressType)))
+            {
+                if (!progressByType.ContainsKey(progressType))
+                {
+                    progressByType.Add(progressType, 0f);
+                }
+            }
+        }
+
+        public void ResetProgress()
+        {
+            foreach (LoadingProgressType progressType in progressByType.Keys.ToList())
+            {
+                progressByType[progressType] = 0f;
+            }
+        }
+
+        public void SetBarFromScene(Scene scene)
+        {
+            this.bar = scene.GetRootGameObjects().FirstOrDefault(go => go.name == "LoadingSphere").transform.Find("LoadingCanvas/LoadingImage/ProgressBackground/ProgressBar").gameObject;
+        }
+
+        private float CalculateOverallProgress()
+        {
+            float totalProgress = 0f;
+            int numTypes = progressByType.Count;
+
+            foreach (var progressPair in progressByType)
+            {
+                totalProgress += progressPair.Value / numTypes;
+            }
+
+            return totalProgress;
+        }
+
+        private void UpdateLoadingBar()
+        {
+            // Calculate the overall progress based on individual progress values
+            float overallProgress = CalculateOverallProgress();
+
+            // Update the loading bar with the overall progress
+            bar.GetComponent<Image>().fillAmount = overallProgress;
+        }
+
+        public void SetProgress(LoadingProgressType progressType, float progress)
+        {
+            progressByType[progressType] = progress;
+            UpdateLoadingBar();
+        }
+
+        public void AddProgress(LoadingProgressType progressType, float progress)
+        {
+            float newProgress = progressByType[progressType] + progress;
+            progressByType[progressType] = newProgress;
+            UpdateLoadingBar();
+        }
+
+        public void SetMaterialForLoading(Scene scene)
+        {
+            scene.GetRootGameObjects().FirstOrDefault(go => go.name == "LoadingSphere").transform.Find("LoadingCanvas/LoadingImage/ProgressBackground/ProgressBar").gameObject.GetComponent<Image>().material = UIManager.I.LoadingBarMaterial;
```

Comment:
I see your point in referencing the scene elements via path. But please think of:
1. Couldn't it be better to have just the last element and do the find? We have DeepFindChildren already in main branch and it could be used. (It's called GameObject.FindChildRecursive()). Because if we change the path sometime, it will all fall apart.
2. If you really want to stick that way, then please name the paths on top as private const variables. (e.g. you already name ProgressBar path twice, which would benefit from your contanciation :-D )